### PR TITLE
stat/distuv: add chi distribution

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

**Comment by kortschak** [comment]
@romanwerpachowski This fails for reasons that I don't understand. The `Prob`, `Quantile` and `Rand` functions seem good, but very clearly `ExKurtosis`, `Mean` and `Variance` lie.

File: stat/distuv/chi.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
```
Comment (romanwerpachowski) [inline stat/distuv/chi.go:1]: Maybe also add Entropy https://en.wikipedia.org/wiki/Chi_distribution#Entropy and Median ?

Comment (kortschak) [reply stat/distuv/chi.go:1]: No good deed goes unpunished.

**Comment by kortschak** [comment]
PTAL

File: stat/distuv/chi.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+// Mean returns the mean of the probability distribution.
+func (c Chi) Mean() float64 {
+	lg1, _ := math.Lgamma((c.K + 1) / 2)
+	lg, _ := math.Lgamma(c.K / 2)
+	return math.Sqrt2 * math.Exp(lg1-lg)
+}
+
+// Median returns the median of the distribution.
+func (c Chi) Median() float64 {
```
Comment (romanwerpachowski) [inline stat/distuv/chi.go:66]: Won't it break for small c.K?

Comment (kortschak) [reply stat/distuv/chi.go:66]: Would you suggest expanding expression?

Comment (kortschak) [reply stat/distuv/chi.go:66]: Using `math.Sqrt(2 * mathext.GammaIncRegInv(0.5*c.K, 0.5))` gives better accuracy.

Comment (romanwerpachowski) [reply stat/distuv/chi.go:66]: so just replace with a call to Quantile(0.5)?

Comment (kortschak) [reply stat/distuv/chi.go:66]: That's what I did. I suspect that what I was thinking this morning when I didn't add it was that I had already added it by virtue of having `Quantile`.

File: stat/distuv/chi.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+// Mean returns the mean of the probability distribution.
+func (c Chi) Mean() float64 {
+	lg1, _ := math.Lgamma((c.K + 1) / 2)
+	lg, _ := math.Lgamma(c.K / 2)
+	return math.Sqrt2 * math.Exp(lg1-lg)
+}
+
+// Mode returns the mode of the distribution.
```
[the remark this answers is not in the document]
Comment (kortschak) [reply stat/distuv/chi.go:70]: Done

File: stat/distuv/chi.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	if x < 0 {
+		return 1
+	}
+	return mathext.GammaIncRegComp(0.5*c.K, 0.5*(x*x))
+}
+
+// Variance returns the variance of the probability distribution.
+func (c Chi) Variance() float64 {
+	m := c.Mean()
+	return c.K - m*m
```
[the remark this answers is not in the document]
Comment (kortschak) [reply stat/distuv/chi.go:123]: Done

[the remark this answers is not in the document]
Comment (romanwerpachowski) [reply stat/distuv/chi.go:123]: Sorry, I meant flooring c.K - m*m at 0.

**Review by romanwerpachowski — CHANGES_REQUESTED** [review]

File: stat/distuv/chi.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (kortschak) [reply stat/distuv/chi.go]: Fixed

**Review by romanwerpachowski — APPROVED** [review]
Thanks.

**Pull request merged**